src/prefect/tasks/aws/secrets_manager.py

```diff
@@ -38,6 +38,8 @@ def run(self, secret: str = None, credentials: str = None) -> dict:
                 with two keys: `ACCESS_KEY` and `SECRET_ACCESS_KEY` which will be
                 passed directly to `boto3`.  If not provided here or in context, `boto3`
                 will fall back on standard AWS rules for authentication.
+            - **kwargs (dict, optional): additional keyword arguments to pass to the
+                Task constructor
```

Comment:
These keyword arguments aren't for the task constructor, they're for boto.

Rather than taking in `**kwargs`, perhaps this task (and all other aws tasks) should get a `boto_kwargs=None` keyword argument for kwargs to forward to boto. This makes more sense to me, since we might want to do something else with `**kwargs` in the future, it's clearer where each argument is going.

If you want, it'd be nice to add `boto_kwargs` to all the aws tasks, but I'd also be happy to merge if you just added it here alone (I can take care of the rest later). Up to you.

Reply:
Very good point! I will add it in a bit to all the aws tasks. 👍 

Just to be sure, you mean to have a boto_kwargs that is per default set to none and otherwise is a dictionary that is forwarded to the respective boto call as `**boto_kwargs` right?

Reply:
> Just to be sure, you mean to have a boto_kwargs that is per default set to none and otherwise is a dictionary that is forwarded to the respective boto call as `**boto_kwargs` right?

Yep, exactly that. Should probably store it on the `Task` always as a dict so we only worry about `None` checks in the constructor. Thanks!

Reply:
@jcrist, what do you think about the changes?